ouroboros_macro/src/lib.rs

```diff
@@ -1711,3 +1711,18 @@ pub fn self_referencing(attr: TokenStream, item: TokenStream) -> TokenStream {
         Err(err) => err.to_compile_error().into(),
     }
 }
+
+/// Functionality inspired by `Inflector`, reimplemented here to avoid the
+/// `regex` dependency.
+fn to_class_case(s: &str) -> String {
+    s.split('_')
```

Comment:
Assumption made here that the `field_name` values above are always of the format `a_b_c_d`.